### DISCUSSION TITLE
How to get a list of modified fields in JS when editing entry

### DISCUSSION THREAD

**Discussion by michaelrog** [craftcms/cms#10752] [Q&A] [answered]
Is it possible to get a list of modified fields when editing an entry in the CP?

I'm looking to implement a warning (JS alert) on entry save if a particular field has changed? (_"Are you sure you want to update this very important field?"_)

I see `Craft.findDeltaData()`, `Craft.cp.$primaryForm`, etc... but it's not clear exactly how these work together. Are there docs on this somewhere that I missed?

**Comment by brandonkelly** [comment, marked as answer]
Craft JS doesn’t keep a definitive list currently. (The `delta` stuff is only for keeping track of what has changed on the current page, but doesn’t factor in any preexisting changes on the draft.) The fields themselves know, though. So you could fire an Ajax request that loads the draft and checks each of its fields.

```php
use craft\base\Element;
use craft\elements\Entry;

$draft = Entry::find()
    ->draftId($draftId)
    ->provisionalDrafts(null)
    ->anyStatus() // Craft 3
    ->status(null) // Craft 4
    ->one();

$modifiedFields = [];

foreach ($draft->getFieldLayout()->getFields() as $field) {
    $status = $field->getStatus($draft);
    if ($status && $status[0] === Element::ATTR_STATUS_MODIFIED) {
        $modifiedFields[] = $field;
    }
}
```

Or you could just search the DOM for fields with a modified status indicator.
